File: TEK3/DOTNET/DOT_AREA_2017/Modules/action/dealabs/CheckNews.cs
using System;
using System.Collections.Generic;
using System.Net;

namespace Modules.action.dealabs
{
    class CheckNews : Action
    {
        WebClient client = new WebClient();
        Dictionary<string, string> fields = new Dictionary<string, string>();
        Reaction r = null;
        List<string> send = new List<string>();
        List<string> page = new List<string>();
        List<string> toSend = new List<string>();
        string content = null;

        public List<string> getFields()
        {
            return (new List<string>() { { "keyword" } });
        }

        public void setField(Dictionary<string, string> fields)
        {
            this.fields = fields;
            send = findNews();
        }

        public void addReaction(Reaction r)
        {
            this.r = r;
        }

        private List<string> findNews()
        {
            var link = "";
            this.fields["keyword"] = this.fields["keyword"].ToLower();
            string toFind = "https://www.dealabs.com/bons-plans/";

            try
            {
                content = (client.DownloadString("https://www.dealabs.com/?page=1"));
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex);
            }

            int i = 0;
            while (i <= content.Length)
            {
                if (i + toFind.Length <= content.Length && content.IndexOf(toFind, i, toFind.Length) == i)
                {
                    i += toFind.Length;
                    link = toFind;
                    int j = 0;
                    while (content[i] != '"')
                        link += content[i++];
                    page.Add(link);
                }
                i++;
            }

            foreach (string links in page)
            {
                if (links.Contains(this.fields["keyword"]) && !toSend.Contains(links))
                    toSend.Add(links);
            }

            return (toSend);
        }
        public void update()
        {
            if (this.r == null)
                return;

            string message = "";

            toSend = findNews();

            foreach (string links in toSend)
            {
                if (!send.Contains(links))
                {
                    message += links;
                    message += "\r\n";
                    send.Add(links);
                }
            }

            if (message != "")
                r.execute(message);
        }
    }
}   

File: TEK3/Java/Java_area_2017/src/main/java/com/area/modules/action/dealabs/CheckNews.java
package com.area.modules.action.dealabs;

import com.area.modules.Action;
import com.area.modules.Reaction;

import java.net.URL;
import java.net.URLConnection;
import java.util.HashMap;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;
import java.util.Scanner;

public class CheckNews implements Action {
    Map<String, String> fields = new HashMap<String, String>();
    Reaction r = null;
    List<String> send = new ArrayList<>();
    List<String> page = new ArrayList<>();
    List<String> toSend = new ArrayList<>();
    String content = null;
    String toFind = "https://www.dealabs.com/bons-plans/";
    URLConnection connection = null;


    public List<String> getFields() {
        return (new ArrayList<String>() {{
            add("keyword");
        }});
    }
    public void setField(Map<String, String> fields) {
        this.fields = fields;
    }
    public void addReaction(Reaction r) {
        this.r = r;
    }
    public void update() {
        if (this.r == null)
            return ;

        toSend.clear();

        try {
            connection =  new URL("https://www.dealabs.com/?page=1").openConnection();
            Scanner scanner = new Scanner(connection.getInputStream());
            scanner.useDelimiter("\\Z");
            content = scanner.next();
        }catch ( Exception ex ) {
            ex.printStackTrace();
        }

        for (int i = 0; i < content.length(); i++) {
            if (content.regionMatches(i, toFind, 0, toFind.length())) {
                i += toFind.length();
                String link = toFind;
                while (content.charAt(i) != '"')
                    link += content.charAt(i++);
                page.add(link);
            }
        }

        int i = 0;
         for (String link:page) {
             if (send.isEmpty())
                if (++i < 21 && link.contains(this.fields.get("keyword")))
                    toSend.add(link);
                else {
                    for (String isSend : send) {
                        if (!isSend.equals(link) && link.contains(this.fields.get("keyword")))
                            if (++i < 21)
                                toSend.add(link);
                    }
                }
        }

        if (!toSend.isEmpty())
            r.execute();
    }
}
